src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx

```diff
@@ -346,36 +359,44 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
       setValue(key, amount)
 
       let estimatedFees
+      let hasEnoughNativeTokenForGas = false
 
       try {
-        estimatedFees = await estimateFormFees()
-        setValue(SendFormFields.EstimatedFees, estimatedFees)
+        // Make API call on first user input
+        // otherwise, debounce user input and use last call
+        if (inputValue.length === 1) {
+          estimatedFees = await estimateFormFees()
+        } else {
+          estimatedFees = await debouncedEstimateFormFees()
+        }
+
+        if (estimatedFees !== undefined) {
+          setValue(SendFormFields.EstimatedFees, estimatedFees)
+
+          hasEnoughNativeTokenForGas = nativeAssetBalance
+            .minus(estimatedFees.fast.txFee)
+            .isPositive()
+        }
       } catch (e) {
         setValue(SendFormFields.AmountFieldError, 'common.insufficientFunds')
-        setLoading(false)
-
         throw e
-      }
+      } finally {
+        const values = getValues()
+        const hasValidBalance = cryptoHumanBalance.gte(values.cryptoAmount)
 
-      const values = getValues()
-
-      const hasValidBalance = cryptoHumanBalance.gte(values.cryptoAmount)
-      const hasEnoughNativeTokenForGas = nativeAssetBalance
-        .minus(estimatedFees.fast.txFee)
-        .isPositive()
-
-      if (!hasValidBalance) {
-        setValue(SendFormFields.AmountFieldError, 'common.insufficientFunds')
-      } else if (!hasEnoughNativeTokenForGas) {
-        setValue(SendFormFields.AmountFieldError, [
-          'modals.send.errors.notEnoughNativeToken',
-          { asset: feeAsset.symbol },
-        ])
-      } else {
-        // Remove existing error messages because the send amount is valid
-        setValue(SendFormFields.AmountFieldError, '')
+        if (!hasValidBalance) {
+          setValue(SendFormFields.AmountFieldError, 'common.insufficientFunds')
```

Comment:
A regression has been introduced here, likely caused by a race condition with this `setValue` now being in the `finally`:

https://user-images.githubusercontent.com/97164662/169206404-2614ea04-ea0a-4c49-aeac-0cac5a8bcf88.mov

I suspect it's because we are early returning in `if (inputValue === '') {...}`.

Reply:
Still seems to be happening 
https://user-images.githubusercontent.com/17035424/173046050-97b740d8-04e6-4e98-8c76-269d5af954cf.mov